### PR TITLE
etcd-tester: fix, clean up multiple things

### PR DESCRIPTION
**Pull request by gyuho**

### CODE REVIEW

**Comment by gyuho** [comment]
/cc @xiang90 I will squash all when merging. I separated this into small changes for review.

Especially https://github.com/coreos/etcd/pull/5462/commits/9d83b2306e02af0c15f3b9951455cad13398af6f.

I misread the indentation in the previous PR.
And some other clean ups, I have been trying to push.

Please review.

Thanks.

File: tools/functional-tester/etcd-tester/tester.go [hunk 1]
```diff
@@ -40,10 +40,10 @@ func (tt *tester) runLoop() {
 	for _, f := range tt.failures {
 		tt.status.Failures = append(tt.status.Failures, f.Desc())
 	}
-	round := 0
```
Comment (xiang90) [inline tools/functional-tester/etcd-tester/tester.go]: let's keep round. it is more readable.

Comment (gyuho) [inline tools/functional-tester/etcd-tester/tester.go]: ok, will revert this commit.

File: tools/functional-tester/etcd-tester/tester.go [hunk 2]
```diff
@@ -131,53 +139,19 @@ func (tt *tester) runLoop() {
...
-	plog.Printf("%s compacted storage (compact revision %d)", tt.logPrefix(), rev)
-
-	plog.Printf("%s checking compaction (compact revision %d)", tt.logPrefix(), rev)
-	if err := tt.cluster.checkCompact(rev); err != nil {
-		plog.Printf("%s checkCompact error (%v)", tt.logPrefix(), err)
-		if cerr := tt.cleanup(); cerr != nil {
-			return fmt.Errorf("%s, %s", err, cerr)
-		}
-		return err
+func (tt *tester) checkCurrentRevision() error {
```
Comment (xiang90) [inline tools/functional-tester/etcd-tester/tester.go]: updateRevision.

File: tools/functional-tester/etcd-tester/tester.go [hunk 3]
```diff
@@ -131,53 +139,19 @@ func (tt *tester) runLoop() {
...
+	revs, _, err := tt.cluster.getRevisionHash()
+	for _, rev := range revs {
```
Comment (xiang90) [inline tools/functional-tester/etcd-tester/tester.go]: what is this for loop doing?

File: tools/functional-tester/etcd-tester/tester.go [hunk 4]
```diff
@@ -131,53 +139,19 @@ func (tt *tester) runLoop() {
...
+		break
 	}
-
-	plog.Printf("%s confirmed compaction (compact revision %d)", tt.logPrefix(), rev)
-	return nil
+	return err
 }
 
-func (tt *tester) defrag() error {
+func (tt *tester) checkCurrentRevisionHash() (failed bool, err error) {
```
Comment (xiang90) [inline tools/functional-tester/etcd-tester/tester.go]: checkConsistency

File: tools/functional-tester/etcd-tester/tester.go (hunk 3 above)
Comment (gyuho) [inline tools/functional-tester/etcd-tester/tester.go]: `revs` are `map[string]int`, so need to range.

Comment (xiang90) [inline tools/functional-tester/etcd-tester/tester.go]: but this loop does nothing...

Comment (xiang90) [inline tools/functional-tester/etcd-tester/tester.go]: probably just break it out at the first time and also add a comment for why you need to do this.

Comment (gyuho) [inline tools/functional-tester/etcd-tester/tester.go]: Ok this already breaks in the first iteration. Will add a comment.

Comment (xiang90) [inline tools/functional-tester/etcd-tester/tester.go]: oh. i was blind... add a comment will help.

**Comment by gyuho** [comment]
/cc @xiang90 All addressed. PTAL.

Thanks!

**Comment by xiang90** [comment]
lgtm

**Comment by gyuho** [comment]
thanks. merging.

**Pull request merged**